packages/context/src/Callback/Callback.container.tsx

```diff
@@ -6,15 +6,20 @@ import withServices from '../withServices';
 export const onRedirectSuccess = (history: ReactOidcHistory, oidcLogInternal: typeof oidcLog) => (user: User) => {
   oidcLogInternal.info('Successfull login Callback', user);
   if (user.state.url) {
-    history.push(user.state.url);
+    history.replaceCurrent(user.state.url);
   } else {
     oidcLogInternal.warn('no location in state');
   }
 };
 
-export const onRedirectError = (history: ReactOidcHistory, oidcLogInternal: typeof oidcLog) => ({ message }: { message: string }) => {
+export const onRedirectError = (history: ReactOidcHistory, oidcLogInternal: typeof oidcLog, userManager: UserManager) => ({ message }: { message: string }) => {
   oidcLogInternal.error(`There was an error handling the token callback: ${message}`);
-  history.push(`/authentication/not-authenticated?message=${encodeURIComponent(message)}`);
+  if (message === "No matching state found in storage") {
```

Comment:
This is exactly what is causing this problem.
 `oidc-client-js` library saves state in the browser state store before it redirects to OIDC provider. This state is removed by `oidc-client-js` immediately after login redirect. When user presses back button, the browser sends the state key (stored in the browser history) in the request. `oidc-client-js` tries to retrieve the state from the storage using the provided key and fails: https://github.com/IdentityModel/oidc-client-js/blob/a8a70ce68bd07373abd7d33e016a97adc5d204bc/src/OidcClient.js#L120. So when user navigates between different pages using back button, he/she will get an error page instead of going to the previous page.
I am not sure if `oidc-client-js` can throw any other exception in this case, but I did not want to assume this is the only one, so I added this check.
Sorry for the lengthy explanation or if I misunderstood your question.

Reply:
if (message === "No matching state found in storage") { // I really think you should remove this line

I think the subject is a little bit more intricate, we have to define exactly the expected behavior.

Reply:
Made the request change, even though I am not  very comfortable with this.

> I think the subject is a little bit more intricate, we have to define exactly the expected behavior.

Agree. My PR addressing the issue the same way it is implemented in the Keycloak JS adapter: they send login redirect call in case signin redirect callback fails to locate saved state in a browser state store.

Reply:
Thank you @mpelesh, I will make many test with many providers.